app/controllers/cart_products_controller.rb

```diff
@@ -0,0 +1,48 @@
+class CartProductsController < ApplicationController
+  before_action :auth_check
```

Comment:
Are you using this check in other controllers as well? If yes then move the method to application controller.